test/NuGet.Tests.Apex/NuGet.Tests.Apex.Daily/NuGetUITestCase.cs

```diff
@@ -877,6 +877,46 @@ public async Task UninstallPackageFromFSharpFromUI()
             CommonUtility.AssertPackageNotInAssetsFile(VisualStudio, project, TestPackageName, TestPackageVersionV1, Logger);
         }
 
+        [TestMethod]
+        [DataRow(ProjectTemplate.ConsoleApplication)]
+        [DataRow(ProjectTemplate.NetCoreConsoleApp)]
+        [Timeout(DefaultTimeout)]
+        public async Task VerifyPackageRestoreAfterDeletingAssetsJsonFile(ProjectTemplate projectTemplate)
+        {
+            // Arrange
+            await CommonUtility.CreatePackageInSourceAsync(_pathContext.PackageSource, TestPackageName, TestPackageVersionV1);
+
+            NuGetApexTestService nugetTestService = GetNuGetTestService();
+            _pathContext.Settings.SetPackageFormatToPackageReference();
+
+            SolutionService solutionService = VisualStudio.Get<SolutionService>();
+            solutionService.CreateEmptySolution("TestSolution", _pathContext.SolutionRoot);
+            ProjectTestExtension project = solutionService.AddProject(ProjectLanguage.CSharp, projectTemplate, "TestProject");
+            VisualStudio.ClearOutputWindow();
+            solutionService.SaveAll();
+
+            CommonUtility.OpenNuGetPackageManagerWithDte(VisualStudio, Logger);
+            NuGetUIProjectTestExtension uiwindow = nugetTestService.GetUIWindowfromProject(project);
+            uiwindow.InstallPackageFromUI(TestPackageName, TestPackageVersionV1);
+
+            var assetsFilePath = CommonUtility.GetAssetsFilePath(project.FullPath);
+            CommonUtility.WaitForFileExists(new FileInfo(assetsFilePath));
+            File.Delete(assetsFilePath);
+
+            // Act
+            if (projectTemplate.Equals(ProjectTemplate.ConsoleApplication))
+            {
+                CommonUtility.RestoreNuGetPackages(VisualStudio, Logger);
```

Comment:
Just curious, why is there different behavior for different project types?

Reply:
Legacy PR vs SDK PR which has auto restore I assume?

If my assumption is correct, why do we need `             CommonUtility.RestoreNuGetPackages(VisualStudio, Logger);` @CiciLi1?

uiwindow.InstallPackageFromUI should have created a project.assets.json.

Reply:
Yes, SDK PR has auto restore feature. we deleted the project.assets.json file before restoring package.
(The test case is to verify that the deleted “project.assets.json” file can be restored by the restore feature)

Reply:
Automatic restore should not be triggered by deleting an assets file, so when would line 913 become relevant?

Reply:
Thanks a lot for the reminder, we thought the `WaitForAutoRestore()` contains restoring project step. And now we have added "unload and reload projects" to the SDK project to trigger the auto-restore feature. Could you please help to review it?

Reply:
We previously had test cases that triggered the package restoring through `CommonUtility.RestoreNuGetPackages(VisualStudio, Logger);`. The purpose of the new test case is to verify that the package file can be restored through automatic restoration. so we use the "unload and reload projects" to trigger the auto-restore.

Reply:
If the intent of this test is to verify unload/reload handles assets file regeneration, then I think the test should be calling that out in the name.

This is a scenario where I think we'd rather see 2 different test cases. 

You can share set-up code through a tertiary method.

Reply:
Thanks to your suggestion, I have split the test cases into two different scenarios, adjusted the names of the test cases. we wrote a  tertiary method for set-up code(unload and reload project).